clash_lib/src/proxy/transport/websocket/websocket.rs

```diff
@@ -85,6 +85,7 @@ impl AsyncWrite for WebsocketConn {
         Pin::new(&mut self.inner)
             .start_send(message)
             .map_err(map_io_error)?;
+        self.poll_flush(cx)?;
```

Comment:
isn't it the purpose of `poll_flush` being one of the trait protocols? https://docs.rs/tokio/latest/tokio/io/trait.AsyncWrite.html#tymethod.poll_flush

Reply:
Yes, but in websocket, when you call the start_send method, you must call the poll_flush method before the network will send this data, otherwise it will not be sent. You can test it in the proxy chain, which includes websocket.

Reply:
i think it's tested via https://github.com/Watfaq/clash-rs/blob/master/clash/tests/data/config/rules.yaml#L63

does it not work for you?

Reply:
yes, it is not work. 
this is my test config file.

![image](https://github.com/Watfaq/clash-rs/assets/149924679/af9a3bf0-131f-4ad9-9456-3f1cf9a2f229)

if you do not poll_flush, This is a screenshot after capturing the packet through wireshark
![image](https://github.com/Watfaq/clash-rs/assets/149924679/ab67d1e1-c70b-48d6-b1e3-df7369d393ed)
this is screenshot after poll_flush. By the way, my initial relay was ss-vmess, but ss didn't seem to work.
![image](https://github.com/Watfaq/clash-rs/assets/149924679/35da6b0b-b401-4402-b845-585afc605aab)

Reply:
i'm not following, is it "ss not working" or "relay vmess-ws + ss" not working?

Reply:
I didn't express myself clearly. I tested ss alone and it didn't work.

Reply:
And similarly, because ss does not work, of course it will not work if there is ss in the relay. I also captured the packet and tested it. It seems that the network is blocked, just like a certain packet is not flushed.

Reply:
ss is tested via https://github.com/Watfaq/clash-rs/blob/4dd786dd85e6993352333086e1df4030e013011c/clash/tests/data/config/ss.yaml#L46

please confirm it's ss problem of ws flush.

i can't reproduce the issue you are describing

Reply:
I'll retest tomorrow and sort out the description. Thanks for your late night reply.

Reply:
My fault, there is no problem with ss, I set the wrong cipher yesterday. But I still think poll_flush is necessary, When both proxies in relay are vmes-ws, this is my test config file. You can test whether it can be reproduced.
```yaml

proxies:
  - name: "vmess-01"
    type: vmess
    uuid: b831381d-6324-4d53-ad4f-8cda48b30811
    server: 127.0.0.1
    port: 16823
    alterId: 0
    cipher: auto
    network: ws
    ws-opts:
      path: /
      headers:
        Host: 127.0.0.1

  - name: "vmess-02"
    type: vmess
    uuid: b831381d-6324-4d53-ad4f-8cda48b30811
    server: 127.0.0.1
    port: 16822
    alterId: 0
    cipher: auto
    network: ws
    ws-opts:
      path: /
      headers:
        Host: 127.0.0.1


proxy-groups:
  - name: "rr"
    type: relay
    proxies:
      - vmess-01
      - vmess-02
rules:
  - MATCH, rr
```

v2ray(version: 5.12.1) config:
```json
{
  "log": {
    "loglevel": "debug"
  },
  "inbounds": [
    {
      "port": 16823,
      "protocol": "vmess",
      "settings": {
        "clients": [
          {
            "id": "b831381d-6324-4d53-ad4f-8cda48b30811",
            "alterId": 0
          }
        ]
      },
      "streamSettings": {
        "network": "ws"
      }
    },
    {
      "port": 16822,
      "protocol": "vmess",
      "settings": {
        "clients": [
          {
            "id": "b831381d-6324-4d53-ad4f-8cda48b30811",
            "alterId": 0
          }
        ]
      },
      "streamSettings": {
        "network": "ws"
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {}
    }
  ]
}
```